src/main/java/Duke.java

```diff
@@ -1,103 +1,222 @@
-                        printMark(taskList, taskNum - 1);
-                    } else {
-                        printTaskNotFound(taskNum);
-                    }
-                }
-
-                else if (userInput.matches("unmark \\d+")) { // check if format of input is "unmark (integer)"
-                    String[] inputArray = userInput.split(" ");
-                    int taskNum = Integer.parseInt(inputArray[1]);
-                    if (taskNum - 1 < (listLength) && taskNum > 0) {
-                        // check if integer provided exceeds length of taskList & ensure taskNum is positive
-                        printUnmark(taskList, taskNum - 1);
-                    } else {
-                        printTaskNotFound(taskNum);
-                    }
-                }
-
-                else { // default add to list case
-                    Task task = new Task(userInput);
-                    taskList[listLength] = task;
-                    listLength++;
-                    printAdd(userInput);
-                }
+    private static void executeMark(String args, boolean hasCompleted) {
+        switch (markCommandVerify(args)) {
+        case "isValid":
+            Task task = taskList[Integer.parseInt(args) - 1];
+            task.setDone(hasCompleted);
+            if (hasCompleted) {
+                printMark(task);
+            } else {
+                printUnmark(task);
             }
+            break;
+        case "isOutOfBounds":
+            invalidMarkArgs(args, true);
+            break;
+        case "isNotNum":
+            invalidMarkArgs(args, false);
+            break;
+        case "noInput":
+            noMarkArgs();
+            break;
+        default:
+            System.out.print(LINE_PARTITION +
+                    "Invalid input for \"mark\" command. Please try again." +
+                    LINE_PARTITION);
         }
     }
-    public static void main(String[] args) {
-        System.out.print(GREET);
-        takeUserInput();
+    private static String markCommandVerify(String details) {
+        if (details.matches("\\d+")){
+            int listNum = Integer.parseInt(details);
+            return (listLength >= listNum && listNum > 0) ? "isValid" : "isOutOfBounds";
+        }
+        return "isNotNum";
+    }
+
+    private static void invalidMarkArgs(String args, boolean isNum) {
+        if (isNum) {
+            System.out.print(LINE_PARTITION +
+                    String.format("\t  Task number %s does not exist.\n", args));
+        } else {
+            System.out.print(LINE_PARTITION +
+                    String.format("\t  %s is not a number.\n", args));
+        }
+        System.out.print(String.format("\t  Please use a number between 1 and %d for \"mark\" and \"unmark\" commands\n", listLength) +
+                LINE_PARTITION);
+    }
+    private static void noMarkArgs() {
```

Comment:
The name of this method may be unclear as it does not indicate an action. Maybe add a verb such as print to become printNoMarkArgs. I believe that would make the purpose of the function clearer.